crates/chia-consensus/src/gen/additions_and_removals.rs

```diff
@@ -0,0 +1,234 @@
+) -> Result<(Vec<(Coin, Option<Bytes>)>, Vec<Coin>), ValidationErr>
+where
+    <I as IntoIterator>::IntoIter: DoubleEndedIterator,
+{
+    let mut a = make_allocator(flags);
+    let mut additions = Vec::<(Coin, Option<Bytes>)>::new();
+    let mut removals = Vec::<Coin>::new();
+
+    let mut cost_left = constants.max_block_cost_clvm;
+
+    let (program, backrefs) = node_from_bytes_backrefs_record(&mut a, program)?;
+
+    let args = setup_generator_args(&mut a, block_refs)?;
+    let dialect = ChiaDialect::new(flags);
+
+    let Reduction(clvm_cost, mut all_spends) =
+        run_program(&mut a, &dialect, program, args, cost_left)?;
+
+    subtract_cost(&a, &mut cost_left, clvm_cost)?;
+    all_spends = first(&a, all_spends)?;
+
+    let mut cache = HashMap::<NodePtr, TreeHash>::new();
+    // at this point all_spends is a list of:
+    // (parent-coin-id puzzle-reveal amount solution . extra)
+    // where extra may be nil, or additional extension data
+
+    while let Some((spend, tail)) = a.next(all_spends) {
+        all_spends = tail;
+        // process the spend
+        let (parent_id, (puzzle, (amount, (solution, _spend_level_extra)))) =
+            <(Bytes32, (NodePtr, (u64, (NodePtr, NodePtr))))>::from_clvm(&a, spend)
+                .map_err(|_| ValidationErr(spend, ErrorCode::InvalidCondition))?;
+
+        let Reduction(clvm_cost, mut iter) =
+            run_program(&mut a, &dialect, puzzle, solution, cost_left)?;
+
+        subtract_cost(&a, &mut cost_left, clvm_cost)?;
+
+        let puzzle_hash = tree_hash_cached(&a, puzzle, &backrefs, &mut cache);
+
+        let coin = Coin {
+            parent_coin_info: parent_id,
+            puzzle_hash: puzzle_hash.into(),
+            amount,
+        };
+
+        removals.push(coin);
+        let spend_id = coin.coin_id();
+
+        while let Some((mut c, next)) = next(&a, iter)? {
+            iter = next;
+            let op = first(&a, c)?;
+            let Ok(op) = atom(&a, op, ErrorCode::InvalidConditionOpcode) else {
+                // unknown opcodes (including pairs) are simply ingnored in
+                // consensus mode
+                continue;
+            };
+            // CREATE_COIN
+            if op.as_ref() != [51_u8] {
+                continue;
+            }
+            c = rest(&a, c)?;
+
+            let (puzzle_hash, (amount, hint)) = <(Bytes32, (u64, NodePtr))>::from_clvm(&a, c)
+                .map_err(|_| ValidationErr(c, ErrorCode::InvalidCondition))?;
+
+            let coin = Coin {
+                parent_coin_info: spend_id,
+                puzzle_hash,
+                amount,
+            };
+
+            // there was another item in the list
+            // the item was a cons-box, and params is the left-hand
+            // side, the list element
+
+            let hint =
+                if let Ok(((hint, _), _)) = <((Bytes, NodePtr), NodePtr)>::from_clvm(&a, hint) {
+                    if hint.len() <= 32 {
```

Comment:
A hint isn't valid unless it's exactly 32 bytes, so you could just use `Bytes32` and skip this check here right?

Reply:
the first incarnation of this patch required it to be exactly 32 bytes, but the existing rules (in `run_block_generator()`) is that as long as it's <= 32 bytes, we still pass it on to python. So, this is required to stay consistent with that behavior. I discovered this in the unit test where there is such a coin created.